### ISSUE TITLE
LLM prompts get interpreted as commands with zsh/prezto

### ISSUE THREAD

**Issue by anku94** [bakks/butterfish#19]
Hi - just started using this tool and absolutely love it!

It seems to have a minor issue working with [prezto](https://github.com/sorin-ionescu/prezto) and zsh. The prompts (commands starting with uppercase) are interpreted as regular shell commands with that plugin enabled.

I know it's prezto because:

1. It's happening on both Mac and Ubuntu
2. On both these systems it's working fine with bash and stock zsh

```
mymachine ~ ❯❯❯ 🐠 Is this thing on?
zsh: no matches found: on?
mymachine ~ ❯❯❯ 🐠 echo $PS1
%{%}${SSH_TTY:+"%F{9}%n%f%F{7}@%f%F{3}%m%f "}%F{4}${_prompt_sorin_pwd}%(!. %B%F{1}#%f%b.)${editor_info[keymap]} 🐠%{ %?%}
```

Is there an easy way to fix this? Happy to provide more debugging info if needed - thank you!

**Comment by bakks** [comment]
Ok @anku94 I think I have this working.

Issue: Butterfish monitors for child processes of the shell process. If there is a running child process we don't intercept the terminal input, we just forward it. This is to prevent Butterfish from doing weird things while you have processes like vim open with it, but it means that if there is a child process open you can't do things like prompting. Apparently when using prezto there is another zsh child shell running, I don't know why.

Fix: I'm filtering out zsh, bash, and sh from the process names that are considered children. This could potentially have an impact if you run something like `bash -c 'sleep 5'`. Will try this out and see if anything breaks.

Commit: f98c52a47bdf774b52e18b437e8315e483c9a1b8


<img width="502" alt="Screenshot 2023-10-13 at 1 55 14 PM" src="https://github.com/bakks/butterfish/assets/1172710/27d9dbf1-cf85-4841-9d59-34d945ccbb32">

**Comment by bakks** [comment]
This fix was released as part of `v0.2.2` today

**Comment by anku94** [comment]
Confirming that this works! I will turn off the prezto autosuggest if it starts becoming a problem, thank you!
